PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,438 @@
+              'dodgerblue', 'turquoise', 'grey', 'gold', 'pink']
+    nextColor = 0
+
+    def __init__(self, size=12, title="Priority Queue", **kwargs):
+        super().__init__(title=title, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = []
+        self.buttons = self.makeButtons()
+
+        
+        self.display()
+
+    def __str__(self):
+        return str(self.list)
+
+    # ARRAY FUNCTIONALITY
+
+    def createIndex(self, index,  # cell
+                    name=None,  # with an optional name label
+                    level=1,  # at a particular level away from the cells
+                    color=None):  # (negative are below)
+        if not color: color = self.VARIABLE_COLOR
+
+        cell_coords = self.cellCoords(index)
+        cell_center = self.cellCenter(index)
+        level_spacing = self.VARIABLE_FONT[1]
+        x = cell_center[0]
+        if level > 0:
+            y0 = cell_coords[1] - self.CELL_SIZE * 3 // 5 - level * level_spacing
+            y1 = cell_coords[1] - self.CELL_SIZE * 3 // 10
+        else:
+            y0 = cell_coords[3] + self.CELL_SIZE * 3 // 5 - level * level_spacing
+            y1 = cell_coords[3] + self.CELL_SIZE * 3 // 10
+        arrow = self.canvas.create_line(
+            x, y0, x, y1, arrow="last", fill=color)
+        if name:
+            label = self.canvas.create_text(
+                x + 2, y0, text=name, anchor=SW if level > 0 else NW,
+                font=self.VARIABLE_FONT, fill=color)
+        return (arrow, label) if name else (arrow,)
+
+    def insert(self, val):
+        callEnviron = self.createCallEnvironment()
+        self.startAnimations()
+        j = self.find(val)  # Find where item should go
+
+        self.list.append(drawable(None))
+        
+        indexK = self.createIndex(len(self.list) -1, 'k', level=-2) # create "k" arrow
+        callEnviron |= set(indexK)  
+        
+        for k in range(len(self.list) - 1, j, -1):  # Move bigger items right
+            
+            self.wait(1)
+            for item in indexK:
+                self.canvas.move(item, -self.CELL_SIZE, 0)  # Move "k" arrow  
+                
+            self.list[k].val = self.list[k - 1].val 
+            self.assignElement(k - 1, k, callEnviron)
+            
+        # Location of the new cell in the array
+        toPositions = (self.cellCoords(j),
+                       self.cellCenter(j))
+
+        # Animate arrival of new value from operations panel area
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        startPosition = [canvasDimensions[0] // 2, canvasDimensions[1]] * 2
+        startPosition = add_vector(startPosition, (0, 0, self.CELL_SIZE, self.CELL_SIZE))
+        cellPair = self.createCellValue(startPosition, val)
+        self.moveItemsTo(cellPair, toPositions, steps= self.CELL_SIZE, sleepTime=0.01)
+        self.list[j]= (drawable(
+            val, self.canvas.itemconfigure(cellPair[0], 'fill'), *cellPair))
```

Comment:
The `cellPair` created here is "temporary" until it is added to the array.  The user could stop the animation during the movement, so we have mark the items in the cellPair as temporary by putting them in the callEnviron and then taking them back out once they should remain.  Specifically:
```
        cellPair = self.createCellValue(startPosition, val)
        callEnviron |= set(cellPair)  # Mark new cell as temporary
        self.moveItemsTo(cellPair, toPositions, steps= self.CELL_SIZE, sleepTime=0.01)
        callEnviron -= set(cellPair)  # New cell is no longer temporary
        self.list[j]= (drawable(
            val, self.canvas.itemconfigure(cellPair[0], 'fill'), *cellPair))
```